File: app/src/test/java/com/usharik/app/AppStateTest.java
package com.usharik.app;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class AppStateTest {

    @Test
    public void defaultsExposeOnlyGlobalSettingValues() {
        AppState appState = new AppState();

        assertEquals(Gender.ALL, appState.getGenderFilterStr());
        assertEquals(-1, appState.getGenderFilterId());
        assertFalse(appState.getSwitchOffAnimation());
        assertEquals(0, appState.getWordsCountSinceLastAd());
        assertTrue(appState.getWordsWithErrors().isEmpty());
    }
}

File: app/src/test/kotlin/com/usharik/app/AppStateTest.kt
package com.usharik.app

import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class AppStateTest {
    @Test fun defaultsAreSafeForANewInstallation() {
        AppState().also { state ->
            assertEquals(Gender.ALL, state.getGenderFilterStr())
            assertFalse(state.getSwitchOffAnimation())
            assertTrue(state.getWordsWithErrors().isEmpty())
            assertEquals(AppState.DEFAULT_HANDBOOK_GENDER, state.getHandbookGender())
            assertEquals(mapOf(AppState.DEFAULT_HANDBOOK_GENDER to AppState.DEFAULT_HANDBOOK_PARADIGM), state.getHandbookParadigmByGender())
        }
    }

    @Test fun handbookGenderIsPersisted() {
        AppState().also { state ->
            state.setHandbookGender("FEMININE")
            assertEquals("FEMININE", state.getHandbookGender())
        }
    }

    @Test fun handbookParadigmIsPersistedPerGender() {
        AppState().also { state ->
            state.setHandbookParadigm("MASCULINE", "hrad")
            state.setHandbookParadigm("FEMININE", "růže")
            assertEquals("hrad", state.getHandbookParadigmByGender()["MASCULINE"])
            assertEquals("růže", state.getHandbookParadigmByGender()["FEMININE"])
        }
    }
}
